src/services/vehicleLoader.ts

```diff
@@ -0,0 +1,39 @@
+import { register } from "./actions";
+import { getDistanceFromProgress } from "./spacingEditor";
+import { forEachVehicle, VehicleSpan } from "./vehicleSpan";
+
+const execute = register<LoadVehicleArgs>("rve-load-car", loadVehicleSettings);
+
+export interface VehicleLoadSettings
+{
+    trackLocation:        CarTrackLocation;
+	trackProgress:        number;
+}
+
+interface LoadVehicleArgs
+{
+    targets:  VehicleSpan[];
+    settings: VehicleLoadSettings[];
+}
+
+export function loadVehicle(targets: VehicleSpan[], settings: VehicleLoadSettings[]): void
+{
+    execute({targets, settings});
+}
+
+function loadVehicleSettings(args: LoadVehicleArgs): void
+{
+    let c = 0;
+    forEachVehicle(args.targets, (car) => {
```

Comment:
Please put braces on the next line according to the code style in the rest of the project.